platform/ios/MGLMapView.mm

```diff
@@ -172,9 +172,18 @@ - (void)setStyleJSON:(NSString *)styleJSON
     }
 }
 
-- (void)setStyleURL:(NSString *)filePathURL
+- (void)setStyleURL:(NSString *)styleURLString
 {
-    mbglMap->setStyleURL(std::string("asset://") + [filePathURL UTF8String]);
+    std::string styleURL([styleURLString UTF8String]);
+
+    if ( ! [[NSURL URLWithString:styleURLString] scheme])
```

Comment:
`setStyleURL:` is private right now. At such time that we support in-framework setting of a style URL (versus download of JSON externally and passing *that*) I think we could, but also, it's weird to use `asset://` — this has no meaning outside of GL and typically a Cocoa dev would use `+[NSURL fileURLWithPath]`.